tests/xdp/xdp_tests.cpp

```diff
@@ -117,31 +52,219 @@ TEST_CASE("xdp_reflect_test", "[xdp_tests]")
+    int sock_type;
+    int protocol;
+    WSAOVERLAPPED overlapped;
+    std::vector<char> recv_buffer;
+    uint32_t recv_flags;
+    sockaddr_storage sender_address;
+    int sender_address_size;
+    uint32_t bytes_received = 0;
+} receiver_socket_t;
+
+TEST_CASE("xdp_encap_reflect_test", "[xdp_tests]")
+{
+    // Initialize the remote address.
+    struct sockaddr_storage remote_address = {};
+    ADDRESS_FAMILY address_family;
+    _get_remote_address(remote_address, &address_family);
+    REQUIRE((address_family == AF_INET || address_family == AF_INET6));
+    int protocol = (address_family == AF_INET) ? IPPROTO_IPV4 : IPPROTO_IPV6;
+    // Create a RAW receiver socket with protocol being IPV4 or IPV6 based on the address family of the remote host.
+    receiver_socket_t receiver_socket(SOCK_RAW, protocol);
+    // Post an asynchronous receive on the receiver socket.
+    receiver_socket.post_async_receive();
+    // send message to remote host on reflection port.
+    const char* message = "Bo!ng";
+    _send_message_to_remote_host(message, remote_address);
+    // Complete the asynchronous receive and obtain the reflected message.
+    receiver_socket.complete_async_receive();
+    // Verify if the received message was reflected by remote host.
+    PSOCKADDR sender_address = nullptr;
+    int sender_length = 0;
+    receiver_socket.get_sender_address(sender_address, sender_length);
+    REQUIRE(INET_ADDR_EQUAL(
+        remote_address.ss_family, INETADDR_ADDRESS((PSOCKADDR)&remote_address), INETADDR_ADDRESS(sender_address)));
+    // Verify if the received message is expected.
+    uint32_t bytes_received = 0;
+    char* received_message = nullptr;
+    receiver_socket.get_received_message(bytes_received, received_message);
+    if (address_family == AF_INET) {
+        // Raw sockets with protocol IPPROTO_IPV4 receives outer IP header in received message.
+        uint32_t remaining_bytes = bytes_received;
+        REQUIRE(remaining_bytes > sizeof(IPV4_HEADER));
+        remaining_bytes -= sizeof(IPV4_HEADER);
+        IPV4_HEADER* inner_ipv4_header = reinterpret_cast<IPV4_HEADER*>(received_message + sizeof(IPV4_HEADER));
+        REQUIRE(remaining_bytes > inner_ipv4_header->HeaderLength * sizeof(uint32_t));
+        remaining_bytes -= inner_ipv4_header->HeaderLength * sizeof(uint32_t);
+        REQUIRE(inner_ipv4_header->Protocol == IPPROTO_UDP);
+        struct _udp_header* udp_header = reinterpret_cast<struct _udp_header*>(
+            reinterpret_cast<char*>(inner_ipv4_header) + inner_ipv4_header->HeaderLength * sizeof(uint32_t));
+        REQUIRE(remaining_bytes > sizeof(struct _udp_header));
+        remaining_bytes -= sizeof(struct _udp_header);
+        REQUIRE(remaining_bytes == strlen(message));
+        REQUIRE(memcmp(reinterpret_cast<char*>(udp_header + 1), message, strlen(message)) == 0);
+    } else {
+        // Raw sockets with protocol IPPROTO_IPV4 does not receive outer IP header in received message.
```

Comment:
```suggestion
        // Raw socket with protocol IPPROTO_IPV4 did not receive outer IP header in received message.
```
Should "IPV4" be "IPV6" in the above comment or is the IPV4 intentional?

Reply:
It should be IPv6.